synapse/pipeline.yml

```
@@ -527,3 +527,39 @@ steps:
 #
 #  - label: Trigger webhook
 #    command: "curl -k https://coveralls.io/webhook?repo_token=$COVERALLS_REPO_TOKEN -d \"payload[build_num]=$BUILDKITE_BUILD_NUMBER&payload[status]=done\""
+
+  ################################################################################
+  #
+  # Complement Test Suite
+  #
+  ################################################################################
+
+  - command:
+      # Build a docker image from the checked out Synapse source
+      - "docker build -t matrixdotorg/synapse:latest -f docker/Dockerfile ."
+      # Download Complement's source code
+      - "wget -N https://github.com/matrix-org/complement/archive/master.tar.gz && tar -xzf master.tar.gz"
+      # Build a second docker image on top of the above image. This one sets up Synapse with a generated config file,
+      # signing and SSL keys so Synapse can run and federate
+      - "docker build -t complement-synapse -f complement-master/dockerfiles/Synapse.Dockerfile complement-master/dockerfiles"
+      # Finally, run the tests. Complement has already been built in the matrixdotorg/complement:latest" image which we're running in
```

well no it hasn't, and even if it had, we're not using the version from the image; we're using the version we downloaded. `go test` does the compilation automatically.

```suggestion
      # Finally, compile and run the tests.
```